calliope/preprocess/model_data.py

```diff
@@ -9,17 +9,21 @@
 time-varying param_dict.
 
 """
-import ast
+import os
+import re
+import logging
 
-import ruamel.yaml
 import xarray as xr
 import numpy as np
 import pandas as pd
 
+import calliope
+from calliope import exceptions
 from calliope.core.attrdict import AttrDict
 from calliope._version import __version__
 from calliope.preprocess import checks
-from calliope.preprocess.time import add_time_dimension, update_dtypes
+from calliope.preprocess import time
+from calliope.core.util import dataset
 
 
 def build_model_data(model_run, debug=False):
```

Comment:
I like the new class-based structure. To be consistent, I'd kill this function too by moving its functionality into `ModelData.init` . The entire interface to the module (and class) from outside the module would then be:

```Python
from calliope.preprocess.model_data import ModelData

model_data = ModelData(model_run)
model_data_original = model_data.model_data_original
model_data = model_data.model_data # FIX name conflict
if debug:
    data_pre_time = model_data.data_pre_time
```

This interface shows, however, the name `model_data` is slightly overstretched. Anyways, I would not worry to much about that at this point.